docs/dev/update-go.md

```diff
@@ -0,0 +1,12 @@
+# Updating the Golang Version for TTPForge
+
+---
+
+If you want to change the supported/recommended Golang version(s) for TTPForge,
+you must do all of the following:
+
+* Update minimum required Golang version specified in `go.mod`
+* Make a corresponding update to `magefiles/go.mod`
```

Comment:
I believe a couple of our other issues are due to inconsistencies that crop up here, so I documented everywhere that consistency is needed